File: src/LC124.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func maxPathSum(root *TreeNode) int {

}

func dfs(root * TreeNode) int{
	l := dfs(root.left)
	r := dfs(root.right)
	if l + root.val > r + root.val{
		return l + root.val
}
	else{
		return r + root.val
}
}

File: src/LC124.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int res;
    vector<int> vec;
    int maxPathSum(TreeNode* root) {
        vec.emplace_back(0);
        vec.emplace_back(0);

        res = root->val;
        dfs(root)[0];
        return res;
    }

    // [leftmax, rightmax]
    void dfs(TreeNode* node) {
        if (nullptr == node) {
            vec[0] = -100000000;
            vec[1] = -100000000;
        }
        auto left = dfs(node->left);
        auto right = dfs(node->right);

        vec[0] = max(max(left[0], left[1]), 0) + node->val;
        vec[1] = max(max(right[0], right[1]), 0) + node->val;
        res = max(res, vec[0] + vec[1] - node->val);
    }
};